scripts/extension-version.test.js

```diff
@@ -0,0 +1,149 @@
+const { expect } = require('chai')
+
+const { nextExtensionVersion } = require('./extension-version')
+
+describe('next extension version', () => {
+  it('it should work with stable channel and current version to new version schema', () => {
+    expect(
+      nextExtensionVersion({
+        prismaVersion: '2.0.0-beta.3',
+        extensionVersion: '0.0.35',
+      }),
+    ).to.eq('0.1.3')
+  })
+
+  it('it should work with stable channel and extension only push', () => {
+    expect(
+      nextExtensionVersion({
+        prismaVersion: '2.0.0-beta.3',
+        extensionVersion: '0.0.3',
+      }),
+    ).to.eq('0.1.3.1')
+  })
+
+  it('it should work with stable channel and new prisma version', () => {
+    expect(
+      nextExtensionVersion({
+        prismaVersion: '2.0.0-beta.4',
+        extensionVersion: '0.0.3',
+      }),
+    ).to.eq('0.1.4')
+  })
+
+  it('it should work with stable channel and new prisma version after an extension only publish', () => {
+    expect(
+      nextExtensionVersion({
+        prismaVersion: '2.0.0-beta.4',
+        extensionVersion: '0.0.3.1',
+      }),
+    ).to.eq('0.1.4')
+  })
+
+  it('it should work with unstable channel and current version to new version schema', () => {
+    expect(
+      nextExtensionVersion({
+        prismaVersion: '2.0.0-alpha.1149',
+        extensionVersion: '0.0.38',
+      }),
+    ).to.eq('0.0.1149')
```

Comment:
Semi offtopic: As the `2` is fixed, couldn't we just prefix the first "major" (here `0`) with `2` to make it "prettier" and more relatable to the real version?

Reply:
Depends on if we expect the minor version of stable/unstable channel in Prisma version to change from 0 to 1. If that happens, we lose that information don't we? 

Right now, we consciously lose the major information knowing that it will be `2` for very long. We can keep it and lose the minor part instead.

Reply:
> Depends on if we expect the minor version of stable/unstable channel in Prisma version to change from 0 to 1. If that happens, we lose that information don't we?

Huh? If I prefix `0` with `2` I get `20`.

Reply:
Ah, got it, example would have helped. We can prefix yes. The output version feels weird though, starting with a 20 (makes it feel like ultra stable).

Reply:
Please create a Slack thread about this suggestion and see what others think - just dropping the `2` altogether for the unstable channel or just concatenating the major and minor together.